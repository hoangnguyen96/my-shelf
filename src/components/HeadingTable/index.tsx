import { memo } from "react";
import { Flex, Text } from "@chakra-ui/react";

interface HeadingTableProps {
  isStatus?: boolean;
}

const HeadingTable = ({ isStatus = true }: HeadingTableProps) => (
  <Flex
    gap={{ base: "2%", xl: "5%", "2xl": "6%" }}
    alignItems="center"
    bgColor="backgroundContent"
    pos="sticky"
    w="100%"
    minH="48px"
    top={0}
    minW={isStatus ? 740 : 700}
    zIndex={10}
  >
    <Text
      size="xl"
      minW={{ base: 227, xl: 260, "2xl": 318 }}
      maxW={{ base: 227, xl: 260, "2xl": 318 }}
      fontWeight={500}
      gap={{ base: "20px", "2xl": "47px" }}
    >
      Title
    </Text>
    <Flex
      alignItems="center"
      minW={{ base: 228, "2xl": 296 }}
      maxW={{ base: 228, "2xl": 296 }}
      gap={{ base: "10px", xl: "20px", "2xl": "60px" }}
    >
      <Text size="xl" fontWeight={500}>
        Ratings
      </Text>
      <Text size="xl" fontWeight={500}>
        Category
      </Text>
    </Flex>
    {isStatus && (
      <Flex maxW={{ base: 100, "2xl": 200 }} w="100%">
        <Text size="xl" fontWeight={500}>
          Status
        </Text>
      </Flex>
    )}
  </Flex>
);

export default memo(HeadingTable);
