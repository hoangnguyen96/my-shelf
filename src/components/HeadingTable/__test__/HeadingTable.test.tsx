import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import HeadingTable from "..";

describe("HeadingTable", () => {
  it("Should render correctly snapshot", () => {
    expect(render(<HeadingTable />)).toMatchSnapshot();
  });
});
