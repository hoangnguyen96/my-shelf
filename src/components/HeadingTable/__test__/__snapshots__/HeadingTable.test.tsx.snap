// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingTable Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-10erws7"
      >
        <p
          class="chakra-text css-1l4rx3f"
        >
          Title
        </p>
        <div
          class="css-bxl9jb"
        >
          <p
            class="chakra-text css-10rvbm3"
          >
            Ratings
          </p>
          <p
            class="chakra-text css-10rvbm3"
          >
            Category
          </p>
        </div>
        <div
          class="css-pelxgl"
        >
          <p
            class="chakra-text css-10rvbm3"
          >
            Status
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-10erws7"
    >
      <p
        class="chakra-text css-1l4rx3f"
      >
        Title
      </p>
      <div
        class="css-bxl9jb"
      >
        <p
          class="chakra-text css-10rvbm3"
        >
          Ratings
        </p>
        <p
          class="chakra-text css-10rvbm3"
        >
          Category
        </p>
      </div>
      <div
        class="css-pelxgl"
      >
        <p
          class="chakra-text css-10rvbm3"
        >
          Status
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
